components/globals/ErrorBoundary.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { Component, ErrorInfo, ReactNode } from "react";
+import { useRouter } from "next/router";
+import { ErrorPanel } from "./ErrorPanel";
+
+interface Props {
+  children?: ReactNode;
+  fallback?: ReactNode;
+}
+
+interface State {
+  hasError: boolean;
+}
+
+// use for testing error boundary
+export const TriggerError = () => {
+  const router = useRouter();
+  const { forceError } = router.query;
+
+  if (forceError) {
+    // Simulate a JS error
+    throw new Error("I crashed!");
+  }
+  return null;
+};
+
+export class ErrorBoundary extends Component<Props, State> {
+  public state: State = {
+    hasError: false,
+  };
+
+  public static getDerivedStateFromError(): State {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true };
+  }
+
+  public componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    /* eslint-disable no-console */
+    console.error("Uncaught error:", error, errorInfo);
```

Comment:
Future thought: I wonder if this would be worth logging?

Reply:
@bryan-robitaille -- thoughts --- logging might be the way to go here but not sure how noisy it will be and or if we'll have eyes on the log for this.

Reply:
In theory, we could log it as a `logMessage.warn()` and change the Pino logger level to `warn` for the client.  But in practice, we've had a lot of difficulty getting that to work properly.  There is also the issue of potentially including sensitive/protected information in the error and stack trace.  I honestly think that we'll be removing the client-side logging in the future.  We may however want to display the error as part of the ErrorPanel so that a user can copy/paste the error if they choose when they contact support?

Reply:
Will adjust as need be in a follow-up.